src/main/java/com/facebook/openwifirrm/modules/ProvMonitor.java

```diff
@@ -105,25 +110,41 @@ private void runImpl() {
 		// ignoring "entity" completely
 		InventoryTagList inventory = client.getProvInventory();
 		SerialNumberList inventoryForRRM = client.getProvInventoryForRRM();
-		VenueList venueList = client.getProvVenues();
 		//EntityList entityList = client.getProvEntities();
 		if (inventory == null || inventoryForRRM == null) {
 			logger.error("Failed to fetch inventory from owprov");
 			return;
 		}
+
+		// fetch the RRM details for each AP that has RRM enabled
+		Map<String, RRMDetails> rrmDetails = new HashMap<String, RRMDetails>();
+		for (String serialNumber : inventoryForRRM.serialNumbers) {
+			RRMDetails details =
+				client.getProvInventoryRrmDetails(serialNumber);
+			if (details == null) {
+				logger
+					.error("Could not fetch RRM details for {}", serialNumber);
+				continue;
+			}
+
+			rrmDetails.put(serialNumber, details);
+		}
+
+		VenueList venueList = client.getProvVenues();
```

Comment:
My question is that, will `getProvVenues` affect fetching RRM details process? If `getProvVenues` failed, is it expected to early exit the method?

Reply:
So in this implementation, we don't technically need this call. However, the "proper" implementation would be that prov would return RRM schedules per venue, not resolved config per device. So if this fails in that implementation, then it makes sense to exit the method early

Reply:
If it makes sense to exit the method early, maybe putting these lines before fetching RRM details process?

Reply:
^ I would follow this suggestion and move the `VenueList` block up.